File: Shows/current/java/Factory/Show/src/DessertFactory.java
public class DessertFactory {
    public Dessert getDessert(String dessertType) {
        if (dessertType == null) {
            return null;
        }
        if (dessertType.equalsIgnoreCase("CAKE")) {
            return new Cake();
        } else if (dessertType.equalsIgnoreCase("ICECREAM")) {
            return new IceCream();
        }
        return null;
    }
}

File: PHP/Creational/Factory/Example3/DessertFactory.php
<?php

include_once 'Cake.php';
include_once 'IceCream.php';

class DessertFactory
{    public static function createDessert(string $type): ?Dessert {
        switch (strtolower($type)) {
            case 'cake':
                return new Cake();
            case 'icecream':
                return new IceCream();
            default:
                return null;
        }
    }
}